src/content/docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/monitor-services/monitor-services-running-kubernetes.mdx

```diff
@@ -0,0 +1,342 @@
+
+```yaml
+# Top level name can be arbitrary, akin to a config file name
+redis-sampleapp.yaml:
+  # Discovery section will define which pods will be monitored.
+  discovery:
+    command:
+      # nri-discovery-kubernetes is a tool that connects to the Kubelet to fetch local pods
+      # without putting stress in the API Server. It accepts the following arguments:
+      # --namespaces: Comma separated namespaces to limit discovery on
+      # --tls: Use tls for connecting to the kubelet
+      # --port: Port used to connect to the kubelet. Defaults to 10255.
+      exec: /var/db/newrelic-infra/nri-discovery-kubernetes --tls --port 10250
+    # Monitor pods which have a `app=sampleapp` label
+    match:
+      label.app: sampleapp
+
+  # Integrations section contains the integration configs.
+  # ${placeholders} will be replaced with the specified attributes for each pod that is discovered
+  integrations:
+    - name: nri-redis  # Integration name, should not be changed
+      env:
+        # Using the discovered pod IP as the hostname address
+        HOSTNAME: ${discovery.ip}
+        PORT: 6379
+        # Other integration options go here
+```
+
+
+
+This snippet must be added to the `integrations` section, under `newrelic-infrastructure` in your `values-newrelic.yaml` file. For example:
+
+```yaml
+global:
+  licenseKey: <var>YOUR_NEW_RELIC_LICENSE_KEY</var>
+  cluster: <var>K8S_CLUSTER_NAME</var>
+
+# Other settings...
+
+newrelic-infrastructure:
+  # verboseLog: true
+  integrations:
+    redis-sampleapp.yaml:
+      discovery:
+        command:
+          # --namespaces: Comma separated list of namespaces to discover pods on
+          # --port: Port used to connect to the kubelet. Default is 10255
+          # --tls: Use secure (TLS) connection
+          exec: /var/db/newrelic-infra/nri-discovery-kubernetes <mark>--tls --port 10250</mark>
+        match:
+          label.app: sampleapp
+
+      integrations:
+        - name: nri-redis
+          env:
+            HOSTNAME: ${discovery.ip}
+            PORT: 6379
+```
+
+<Callout variant="tip">
+  Notice we are specifying `--tls --port 10250`. Earlier versions of the integration might have worked without this, but it is mandatory since v3 as we now connect to the Kubelet using the nodeIP rather than `localhost`.
+</Callout>
+
+Integrations targeting other pods should have their own section next to `redis-sampleapp.yaml`.
+
+Integrations are standalone binaries, and are executed by the infrastructure agent that is included in the `newrelic-nrk8s-kubelet-xxxxx` pods. Config files are deployed to all pods of the `nrk8s-kubelet` DaemonSet, but discovery ensures that each pod will only target service pods that are scheduled in the same node as that particular `nrk8s-kubelet` pod. If an instance of the `nrk8s-kubelet` DaemonSet does not have any pods matching the specified labels, the integration will not be executed by that instance.
```

Comment:
NTH -> A small diagram could the user understanding this section

Reply:
PRs welcome :P

Reply:
No idea if it is simplifying things, what do you think? If so we can ask alba to make it nice enough to be public 😄 
![Screen Shot 2022-01-27 at 5 04 38 PM](https://user-images.githubusercontent.com/43335750/151396664-1375803d-39b6-4a9f-a84a-6b14c74374c5.png)

Reply:
https://drive.google.com/file/d/1Y1e4AZIB79dpRdu0cvMhMz_0EZgg_3_2/view?usp=sharing